spec/system/consumer/split_checkout_spec.rb

```diff
@@ -242,6 +261,31 @@
       end
     end
 
+    context "payment method step" do
+      let(:order) { create(:order_ready_for_payment, distributor: distributor) }
+
+      it "preselect the payment method if only one is available" do
+        visit checkout_step_path(:payment)
+
+        expect(page).to have_checked_field "payment_method_#{payment_method.id}"
+      end
+
+      context "with more than one payment method" do
+        let!(:payment_method2) { create(:payment_method, distributors: [distributor]) }
+
+        before do
+          order.distributor.update! payment_methods: [payment_method, payment_method2]
+        end
```

Comment:
You don't need to update the distributor because you assigned the payment method already at creation.

```suggestion
        let!(:payment_method2) { create(:payment_method, distributors: [distributor]) }
```